src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt

```diff
@@ -0,0 +1,58 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.sentencing
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.BadDataException
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.NotFoundException
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.toCodeDescription
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.Offender
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderBooking
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.CourtCaseRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.OffenderRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.findRootByNomisId
+
+@Service
+@Transactional
+class SentencingService(
+  private val courtCaseRepository: CourtCaseRepository,
+  private val offenderRepository: OffenderRepository,
+  private val telemetryClient: TelemetryClient,
+) {
+  fun getCourtCase(id: Long, offenderNo: String): CourtCaseResponse {
+    findPrisoner(offenderNo).findLatestBooking()
```

Comment:
this validation of offender/booking may be a. bit overboard here - especially if used in migration.  Leaving in for now as may assist with development/testing